### PR TITLE
CI: separate step to download nltk files

### LINKED ISSUE #32667: `run_summarization` likely race condition creates flaky `test_tensorflow_examples`
As in title. See for instance https://app.circleci.com/pipelines/github/huggingface/transformers/100640/workflows/f4e73659-1f14-45da-8740-7314e788fb72/jobs/1340353 that happened in #32651 . Can't reproduce locally. 
Caused by https://github.com/huggingface/transformers/blob/9d2ab8824c2a820a0ac9f2cd884ef838ca77b481/examples/tensorflow/summarization/run_summarization.py#L62-L70 :

Will cause a failure at test collection

```bash
_______ ERROR collecting examples/tensorflow/test_tensorflow_examples.py _______
examples/tensorflow/test_tensorflow_examples.py:67: in <module>
    import run_summarization
examples/tensorflow/summarization/run_summarization.py:63: in <module>
    nltk.data.find("tokenizers/punkt")
/usr/local/lib/python3.10/site-packages/nltk/data.py:554: in find
    return find(modified_name, paths)
/usr/local/lib/python3.10/site-packages/nltk/data.py:541: in find
    return ZipFilePathPointer(p, zipentry)
/usr/local/lib/python3.10/site-packages/nltk/compat.py:36: in _decorator
    return init_func(*args, **kwargs)
/usr/local/lib/python3.10/site-packages/nltk/data.py:394: in __init__
    zipfile = OpenOnDemandZipFile(os.path.abspath(zipfile))
/usr/local/lib/python3.10/site-packages/nltk/compat.py:36: in _decorator
    return init_func(*args, **kwargs)
/usr/local/lib/python3.10/site-packages/nltk/data.py:943: in __init__
    zipfile.ZipFile.__init__(self, filename)
/usr/local/lib/python3.10/zipfile.py:1271: in __init__
    self._RealGetContents()
/usr/local/lib/python3.10/zipfile.py:1338: in _RealGetContents
    raise BadZipFile("File is not a zip file")
E   zipfile.BadZipFile: File is not a zip file
------------------------------- Captured stderr --------------------------------

```
Would be nice to have: 
- isolate the behaviour of the bad file creation
- flag the flaky part of the failing test accordingly.

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Should fix #32667 : NLTK is complaining about parallelism. The parallelism stems from the `pytest` call, so we should get rid of the problem if we download the NLTK data in advance of the `pytest` call 🤗

### CODE REVIEW

File: .circleci/create_circleci_config.py [hunk 1]
```diff
@@ -121,11 +121,16 @@ def to_dict(self):
...
+        # Examples special case: we need to download NLTK files in advance to avoid cuncurrency issues
+        if "examples" in self.name:
+            steps.append({"run": {"name": "Download NLTK files", "command": """python -c "import nltk; nltk.download('punkt', quiet=True)" """}})
+
         test_command = ""
         if self.command_timeout:
             test_command = f"timeout {self.command_timeout} "
         # junit familiy xunit1 is necessary to support splitting on test name or class name with circleci split
-        test_command += f"python3 -m pytest -rsfE -p no:warnings -o junit_family=xunit1 --tb=short --junitxml=test-results/junit.xml -n {self.pytest_num_workers} " + " ".join(pytest_flags)
+        test_command += f"python3 -m pytest -rsfE -p no:warnings --tb=short -o junit_family=xunit1 --junitxml=test-results/junit.xml -n {self.pytest_num_workers} " + " ".join(pytest_flags)
```
Comment (gante) [inline .circleci/create_circleci_config.py:133]: if we reorder the flags, we can see that this is the same as L188-191 (removed in this PR)

File: .circleci/create_circleci_config.py [hunk 2]
```diff
@@ -121,11 +121,16 @@ def to_dict(self):
         )
 
         steps.append({"run": {"name": "Create `test-results` directory", "command": "mkdir test-results"}})
+
+        # Examples special case: we need to download NLTK files in advance to avoid cuncurrency issues
+        if "examples" in self.name:
+            steps.append({"run": {"name": "Download NLTK files", "command": """python -c "import nltk; nltk.download('punkt', quiet=True)" """}})
```
Comment (amyeroberts) [inline .circleci/create_circleci_config.py:127]: Much nicer solution :)

Comment (ydshieh) [reply .circleci/create_circleci_config.py:127]: nice!

File: .circleci/create_circleci_config.py (hunk 1 above)
Comment (ydshieh) [reply .circleci/create_circleci_config.py:133]: great!

**Review by ydshieh — APPROVED** [review]
You are smart!

**Review by ArthurZucker — APPROVED** [review]

**Pull request merged**